RecoMuon/Configuration/python/RecoMuon_cff.py

```diff
@@ -84,3 +84,6 @@
 
 muonshighlevelrecoTask = cms.Task(muonPFIsolationTask,muons,muonReducedTrackExtras)
 muonshighlevelreco = cms.Sequence(muonshighlevelrecoTask)
+
+from Configuration.ProcessModifiers.pp_on_AA_cff import pp_on_AA
+pp_on_AA.toModify(muons1stStep, minPt = 0.8, selectHighPurity = True)
```

Comment:
I don't think you should retract from the good habit of modifying the modules where those modules are defined (i.e. here in RecoMuon/MuonIdentification/python/muons1stStep_cfi.py)
As you do it now, the configuration of the module `muons1stStep` is different depending whether RecoMuon_cff.py has been already encountered in the configuration or not, which is normally quite unsafe.

I think that what you should do is to move back this modification in muons1stStep_cfi.py, and if needed reset `selectHighPurity` for `pp_on_AA` whenever `muons1stStep` is cloned (there are 14 such cases in CMSSW, not all of them certainly relevant for `pp_on_AA`)

Reply:
moved era customization as suggested in latest commit.